src/coreclr/tools/Common/Compiler/DevirtualizationManager.cs

```diff
@@ -201,5 +201,47 @@ protected virtual MethodDesc ResolveVirtualMethod(MethodDesc declMethod, DefType
 
             return impl;
         }
+
+        /// <summary>
+        /// Check whether compile-time resolution of static virtual method is possible. This returns false for shared
+        /// generics where the resolution needs to be deferred to the runtime working with exact instantiation types.
+        /// </summary>
+        /// <param name="currentType">Type constraint in the static virtual method </param>
+        /// <param name="interfaceMethod">Static virtual interface method to resolve</param>
+        public bool AllowCompileTimeStaticVirtualMethodResolution(TypeDesc currentType, MethodDesc interfaceMethod)
```

Comment:
This shouldn't be just AllowCompileTimeStaticVirtualMethodResolution, we should make the call to ResolveVariantInterfaceMethodToVirtualMethodOnType from within the DevirtualizationManager, as the full and compilex R2R version resilience rules will depend on the result of resolving the virtual call. In particular, it isn't required that the entire base type chain be within the bubble, but that the chain up until a result is found is within the version bubble.

Reply:
Hmm, in such case it seems to me that we should rather include the version bubble edge logic directly in the recursive descent into base type in <code>ResolveVirtualStaticMethod</code> as in contrast to the runtime we only need to succeed the resolution attempt when we stay within the version bubble.

It also seems to me that we only need the variant form of the resolution as the only place where we're using the non-variant form in the runtime is class validity checks and we don't need to be doing that.